### PR TITLE
[ACM-8209] Refactored code for cleanup

### PR DESCRIPTION
**Pull request by dislbenn**
# Description

Refactor code to reduce code smells.

## Related Issue

https://issues.redhat.com/browse/ACM-8209

## Changes Made

Updated code to reduce code smells.

## Screenshots (if applicable)

Add screenshots or GIFs that demonstrate the changes visually, if relevant.

## Checklist

- [ ] I have tested the changes locally and they are functioning as expected.
- [ ] I have updated the documentation (if necessary) to reflect the changes.
- [ ] I have added/updated relevant unit tests (if applicable).
- [ ] I have ensured that my code follows the project's coding standards.
- [ ] I have checked for any potential security issues and addressed them.
- [ ] I have added necessary comments to the code, especially in complex or unclear sections.
- [ ] I have rebased my branch on top of the latest main/master branch.

## Additional Notes

Add any additional notes, context, or information that might be helpful for reviewers.

## Reviewers

Tag the appropriate reviewers who should review this pull request. To add reviewers, please add the following line: `/cc @reviewer1 @reviewer2`

/cc @cameronmwall @ngraham20 

## Definition of Done

- [ ] Code is reviewed.
- [ ] Code is tested.
- [ ] Documentation is updated.
- [ ] All checks and tests pass.
- [ ] Approved by at least one reviewer.
- [ ] Merged into the main/master branch.

### CODE REVIEW

File: controllers/hosted.go [hunk 1]
```diff
@@ -334,32 +332,36 @@ func (r *MultiClusterHubReconciler) calculateHostedStatus(m *operatorv1.MultiClu
 
 	// Copy conditions one by one to not affect original object
```
Comment (ngraham20) [inline controllers/hosted.go:333]: Did you double check this is still true?

Comment (dislbenn) [reply controllers/hosted.go:333]: Yes, I can confirm that this is still true.

**Comment by dislbenn** [comment]
/override sonar-pre-submit
Will override the sonar pre submit job, since the code changes that were made are mostly correcting print statements and removing redundant or unused code.

**Comment by dislbenn** [comment]
/override ci/prow/sonar-pre-submit
Will override the sonar pre submit job, since the code changes that were made are mostly correcting print statements and removing redundant or unused code.

**Comment by dislbenn** [comment]
I will work towards adding more test cases in a separate PR.

**Pull request merged**